lib/OpenQA/Worker/Engines/isotovideo.pm

```diff
@@ -162,6 +162,49 @@ sub cache_assets {
     return;
 }
 
+sub sync_tests {
+    my ($shared_cache) = @_;
+    my $cache_client   = OpenQA::Worker::Cache::Client->new;
+    my $rsync_request  = $cache_client->request->rsync(
+        from => $hosts->{$current_host}{testpoolserver},
+        to   => $shared_cache
+    );
+
+    # TODO: Enqueue all requests in one place
+    return {error => "Failed to send rsync cache request"} unless $rsync_request->enqueue;
+
+    sleep 5 and update_setup_status until $rsync_request->processed;
+    my $exit = $rsync_request->result;
+    log_info("rsync: " . $rsync_request->output, channels => 'autoinst');
+    return $exit;
+}
+
+# do asset caching if CACHEDIRECTORY is set
+sub do_asset_caching {
+    my ($vars) = @_;
+    my $shared_cache;
+    my $assetkeys = detect_asset_keys($vars);
+    if (!$worker_settings->{CACHEDIRECTORY}) {
+        my $error = locate_local_assets($vars, $assetkeys);
+        return $error if $error;
+        return;
```

Comment:
Don't use bare returns. If you want to return `undef` always use `return undef;` to avoid list context bugs.

Reply:
Well, I did not change this return but I will include a commit to change this.

Reply:
Yea, there are many many uses of bare returns in openQA (that should all be changed). I just keep mentioning it since it is one of the few things the book Perl Best Practices got wrong. And because mixed list/scalar context functions have been one of the most common causes of serious security issues in Perl web apps.